Add dataType option to Select fields

JedWatson opened the pull request:
Been meaning to implement this for a long time.

Our Select field should support storing value as an `enum` / `string` / `integer` with the corresponding GraphQL data type. Now it does!

Fixes #851
Fixes #398 
Fixes #1768

MadeByMike commented:
I did a smoke test and admin UI looks ok. I assume this might break if the AdminUI defined a graphQL query that was expecting an enum, and it looks like it doesn't hard code this.

JedWatson commented:
@MadeByMike I think you're right - I have tested the Admin UI but will review the controller again for how filters are submitted.

MadeByMike commented:
This might be overkill but I've added filter tests for each of the data types. All passing.

timleslie commented on packages/fields/src/types/Select/test-fixtures.js:
```diff
@@ -53,13 +71,77 @@ export const filterTests = withKeystone => {
+  );
+
   test(
     'Filter: company',
```
Can we add `(dataType: enum)` on all of the original tests just for consistency?

timleslie commented:
Just from eyeballing the code, it looks to me like `getFilterGraphQL` won't do the right thing when using `dataType = string`. My reading of it is that we should be putting double quotes around `value` in that case.

JedWatson commented:
Fixed the filtering in the Admin UI, added some more helpful type validation for options, and added explicit `enum` case for the tests.

I think this is good to merge now
